### PR TITLE
Adding quest: Caravan site type

### LINKED ISSUE #641: New Quest: What kind of caravan site is this? (`caravan_site:type`)
### General
Affected tag(s) to be modified/added: [caravan_site:type](https://wiki.openstreetmap.org/wiki/Tag:tourism%3Dcaravan_site)
Taginfo: https://taginfo.openstreetmap.org/keys/caravan_site%3Atype#overview
Question asked: **What kind of caravan site is this?**

### Checklist
Checklist for quest suggestions (see [guidelines](https://github.com/Helium314/SCEE/blob/modified/QUEST_GUIDELINES.md)):
- [ ] 🚧 To be added tag is established and has a useful purpose
- [x] 🤔 Any answer the user can give should have an equivalent tagging (SCEE quests may not be answerable in all cases, but please keep this rare)
- [x] 🐿️ A survey is necessary (may require knowledge of the subject, though easy answering for everyone is preferred)
- [x] 💤 Not an overwhelming percentage of quests have the same answer (No spam)
- [x] 🕓 Applies to a reasonable number of map data (Worth the effort)

### Ideas for implementation

**Element selection:** 
```
nodes, ways, relations with
  tourism = caravan_site and (
  !caravan_site:type
)
```

**Possible Answers:** 
- village
- river
- lake
- town
- parking_lot
- harbour
- winery
- camp_site
- museum
- restaurant
- farm
- beach
- supermarket

**Proposed UI:**
Similar to outdoor_seating Quest with a simple text selection of values. Probably not possible without images, but a description can, but does not necessarily have to be added.

### PR DESCRIPTION
**Pull request by mcliquid**
Fixes #641 

- Icon could be improved, but should work
- Strings could also be improved, but they should work out
- Quest was tested

### CODE REVIEW

**Comment by mcliquid** [comment]
FYI: I've fixed the German Term ("Wohnwagen-Stellplatz") for the English Term "RV Park" to "Wohnmobil-Stellplatz".

**Comment by Helium314** [comment]
Looks fine!
I'll have a look at how the icon looks in the app a bit later, judging by the SVG it should be fine though.

Strings should be ok, only thing I noticed is that _harbour_ is British spelling, while the "default English" in Android is supposed to be the US variation.

**Comment by mcliquid** [comment]
Thanks for the quick feedback! I've changed the string to US English.

File: app/src/main/java/de/westnordost/streetcomplete/quests/caravan_site_type/AddCaravanSiteType.kt [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    """
+    override val changesetComment = "Add caravan site type info"
+    override val defaultDisabledMessage = R.string.default_disabled_msg_caravanSiteType
+    override val wikiLink = "Key:caravan_site:type"
+    override val icon = R.drawable.ic_quest_caravan_site
+
+    override fun getTitle(tags: Map<String, String>) = R.string.quest_caravanSiteType_title
+
+    override fun getHighlightedElements(element: Element, getMapData: () -> MapDataWithGeometry) =
+        getMapData().filter("nodes, ways, relations with tourism = caravan_site")
```
Comment (mnalis) [inline app/src/main/java/de/westnordost/streetcomplete/quests/caravan_site_type/AddCaravanSiteType.kt]: maybe also highlight nearby `tourism=camp_site` (all or at least those with `caravans=yes`) ?

Comment (mcliquid) [reply app/src/main/java/de/westnordost/streetcomplete/quests/caravan_site_type/AddCaravanSiteType.kt]: @mnalis Good advice, thank you! I will adapt it. I just recently came across a potential misunderstanding and maybe you can help here.
- `tourism=camp_site` is for caravans (kinda trailers for living)
- `tourism=caravan_site` is for motorhomes / RVs / Camper vans (you stay overnight “in the car” directly)
- `caravans=*` means the trailers for living, not the motorhomes, correct?

Comment (mnalis) [reply app/src/main/java/de/westnordost/streetcomplete/quests/caravan_site_type/AddCaravanSiteType.kt]: Look at the wiki for [tourism=camp_site](https://wiki.openstreetmap.org/wiki/Tag:tourism%3Dcamp_site) for details. 

If I understand correctly, what you call caravans/trailers_for_living is AFAICT tagged with [static_caravans=yes](https://wiki.openstreetmap.org/wiki/Key:static_caravans); and moving ones are either [caravans=yes](https://wiki.openstreetmap.org/wiki/Key:caravans) (for non-motorized separate vehicles to be towed by a car) or [motorhome=yes](https://wiki.openstreetmap.org/wiki/Key:motorhome) (for self-propelled single-vehicle ones, i.e. RVs/motorhomes; if I got nomenclature right :sweat_smile: )

`tourism=camp_site` can be used **for all three categories** (and more, like tents and cabins etc. - but those are not interesting for this PR), depending on those subtags.

Comment (mcliquid) [reply app/src/main/java/de/westnordost/streetcomplete/quests/caravan_site_type/AddCaravanSiteType.kt]: Thanks! I guess I was confused with the access tag `caravan=*`. I've added `camp_site`.

**Review by mnalis — APPROVED** [review]

**Review by mnalis — APPROVED** [review]
should it perhaps have:
```
override val achievements = listOf(OUTDOORS)
```

(or some other achievement?)

**Comment by mcliquid** [comment]
> should it perhaps have:
> 
> ```
> override val achievements = listOf(OUTDOORS)
> ```
> 
> (or some other achievement?)

Can I add to that - I wasn't aware that SCEE quests can also get achievements.

**Comment by goldfndr** [comment]
One more minor edit: "wineyards" should be "vineyards". (I've corrected the wiki page.)

**Pull request merged**